src/twisted/internet/defer.py

```diff
@@ -52,6 +52,12 @@ class TimeoutError(Exception):
     """
 
 
+class InvalidStateError(Exception):
```

Comment:
Another thought about why this should come from ``BaseException``:

There's not really much recovering from this. At best maybe you can run it through the event loop or synchronously drive it and then pass it through again... but if you can do that why didn't you do it in the first place?